Web/Scotty/Action.hs

```diff
@@ -314,24 +338,24 @@ formParam = paramWith FormParam envFormParams status400
 -- * This function raises a code 400 also if the parameter is found, but 'parseParam' fails to parse to the correct type.
 --
 -- /Since: 0.20/
-queryParam :: (Parsable a, Monad m) => T.Text -> ActionT m a
-queryParam = paramWith QueryParam envQueryParams status400
+queryParam :: (Parsable a, MonadIO m) => T.Text -> ActionT m a
+queryParam = paramWith QueryParameterNotFound envQueryParams
 
 -- | Look up a capture parameter. Returns 'Nothing' if the parameter is not found or cannot be parsed at the right type.
 --
 -- NB : Doesn't throw exceptions. In particular, route pattern matching will not continue, so developers
 -- must 'raiseStatus' or 'throw' to signal something went wrong.
 --
 -- /Since: FIXME/
-captureParamMaybe :: (Parsable a, Monad m) => T.Text -> ActionT m (Maybe a)
+captureParamMaybe :: (Parsable a, MonadIO m) => T.Text -> ActionT m (Maybe a)
```

Comment:
why do we need a MonadIO constraint?

Reply:
In order to throw an exception. What's wrong with it?

Reply:
Nothing per se, but if the underlying implementation didn't change why change the type sig?

Reply:
The underlying implementation did change, namely `paramWith` now requires MonadIO because it calls `throwIO` instead of `throw`. In general it's a good practice to avoid throwing an exception in a pure context.

Reply:
@fumieval  ^

Reply:
OK, fixed